### PR TITLE
setup

### PR DESCRIPTION
**Pull request by nicolegerbauld**

### CODE REVIEW

**Comment by nicolegerbauld** [comment]
Hi! I The code is also in my repo: https://github.com/nicolegerbauld/routing.

**Comment by makara** [comment]
Could you ignore the node modules from Git and resend the PR?

**Comment by nicolegerbauld** [comment]
Hi! Just removed the node modules

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+node_modules/
```
Comment (makara) [inline .gitignore]: There are best practice out there, for example, https://github.com/Wiredcraft/api-boilerplate/blob/master/.gitignore, https://github.com/github/gitignore

File: src/ItemDetail.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
+import React, {useState, useEffect} from 'react';
+import './App.css';
+// import {Link} from 'react-router-dom';
+
+function Item() {
+
+  useEffect(() => {
+    fetchItem();
```
Comment (johncalvinroberts) [inline src/ItemDetail.js]: A couple of syntax errors here which are not throwing runtime errors due to Babel.

* calling `fetchItem` before `fetchItem` is defined
* Two variables called `fetchItem`

File: src/ItemDetail.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  },[]);
+
+  const [item, setItem] = useState({});
+
+  const fetchItem = async () => {
+    const fetchItem = await fetch('https://newsapi.org/v2/top-headlines?country=us&apiKey=e6744cfd24904c71b60b673f94f673e2')
+    const item = await fetchItem.json();
```
Comment (johncalvinroberts) [inline src/ItemDetail.js]: What if the request fails? Can you try adding some error handling here?

File: src/Nav.js [hunk 4]
```diff
@@ -0,0 +1,46 @@
+import React from 'react';
+import './App.css';
+import { Link } from 'react-router-dom'
```
Comment (johncalvinroberts) [inline src/Nav.js]: This is what you need to check out ESLint + Prettier for: There is some inconsistent formatting in the codebase.

File: src/Shop.js [hunk 5]
```diff
@@ -0,0 +1,41 @@
+import React, {useState, useEffect} from 'react';
+import './App.css';
+import {Link} from 'react-router-dom';
+
+function Shop() {
```
Comment (johncalvinroberts) [inline src/Shop.js]: Why is this component called `Shop`?

File: src/ItemDetail.js [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+function Item() {
+
+  useEffect(() => {
+    fetchItem();
+  },[]);
+
+  const [item, setItem] = useState({});
+
+  const fetchItem = async () => {
+    const fetchItem = await fetch('https://newsapi.org/v2/top-headlines?country=us&apiKey=e6744cfd24904c71b60b673f94f673e2')
```
Comment (johncalvinroberts) [inline src/ItemDetail.js]: you can get the `id` from `props`, then fetch by id

**Pull request closed without merging**